### DISCUSSION TITLE
Système de sauvegarde / chargement

### DISCUSSION THREAD

**Discussion by uurshin** [fibreville/minotaure_php#57] [Parlons peu, parlons dev]
Actuellement, sans intégration dans un système autour de ATRPG (type celui de @PixelReveur), le seul moyen de garder en mémoire une aventure est de faire un backup de la base (et des fichiers depuis la 2.2). Ca pourrait être pratique de pouvoir sauvegarder / charger une partie mise en pause pour mener plusieurs campagnes.

Je vois plusieurs étapes :
- revoir le login de l'admin pour qu'il ne passe plus par la création de perso et par le système "si id = 1 alors c'est l'admin" mais par un lien et un mécanisme à part (pas pratique pour les requêtes sql). On pourrait avoir plusieurs admin du coup.
- rapatrier tout sauf le mécanisme de "heartbeat" dans base. Une parties des infos en base de données avaient été mis dans des fichiers pour contourner les limites des hébergements gratuit, mais ce n'est plus une éventualité maintenant qu'on sait que ce n'est pas suffisant (test live effectué cette semaine). Le 'heartbeat' c'est un timestamp qui est mis dans un fichier et sert pour savoir s'il y a des infos à récupérer pour les pjs, pour limiter les requêtes mysql inutiles.
- permettre l'export / l'import des données du jeu (peut être sous un autre format que .sql ? un yml/json ou autre format approprié.

**Comment by ElethornGM** [comment]
Je suis d'accord que ça manque de flexibilité et que ce serait un plus de pouvoir faire ça...
